bundler/spec/plugins/install_spec.rb

```diff
@@ -92,16 +92,18 @@
     expect(out).to include("Using foo 1.1")
   end
 
-  it "installs when --branch specified" do
-    bundle "plugin install foo --branch main --source #{file_uri_for(gem_repo2)}"
+  it "raises an error when when --branch specified" do
+    bundle "plugin install foo --branch main --source #{file_uri_for(gem_repo2)}", :raise_on_error => false
```

Comment:
This is another different bug fix / change included with this PR, right? Previously we would interpret `--source` as a `git` source if `branch` or `ref` were specified. Now we're more strict about it.

I'm in general in favor of this since it makes plugin sources more consistent with Gemfile sources, but since this was coded as specs, maybe extract a PR & discuss/review there?

Reply:
I've actually broken this into 4 commits now, but they all depend on each other. I don't know if you have a preference to just have 4 commits in this PR, or one PR per commit. #7045 is the first commit in the stack if you prefer separate PRs; I'll create the next PR once that's merged, etc.